src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/RemovePackageReferenceCommandRunner.cs

```diff
@@ -24,8 +25,8 @@ public Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs, MSBui
             var libraryDependency = new LibraryDependency
             {
                 LibraryRange = new LibraryRange(
-                    name: packageReferenceArgs.PackageDependency.Id,
-                    versionRange: packageReferenceArgs.PackageDependency.VersionRange,
+                    name: packageReferenceArgs.PackageId,
+                    versionRange: packageReferenceArgs.PackageVersion != null ? VersionRange.Parse(packageReferenceArgs.PackageVersion) : VersionRange.Parse("*"),
```

Comment:
For backwards compatibility, I think you want `VersionRange.All`. 
See https://github.com/NuGet/NuGet.Client/blob/9f277be9d55cff0aeb23de18fd62d543a41ff10a/src/NuGet.Core/NuGet.Packaging/Core/PackageDependency.cs#L43 for reference. 

Before the PackageDependency was created using only the id and the default was VersionRange.All.